.circleci/config.yml

```diff
@@ -61,19 +61,108 @@ jobs:
           name: e2e test
           command: |
             set +e
-            if [[ $CIRCLE_BRANCH != *"master"* || ! -z $CIRCLE_TAG ]] ; then
-              E2EPROFILE=saucelight npm run test:e2e
+            npm run test:e2e
+            STATUS=$?;
+            if [[ ! -z $FORCE_BUILD ]] ; then
+              echo "The force build is active! Independently of the tests exit code the build does not fail. The exit code: $STATUS"
+              exit 0
+            else
+              exit $STATUS
+            fi
+
+      - store_artifacts:
+          path: /tmp/circleci-artifacts
+      - store_artifacts:
+          path: /tmp/circleci-test-results
+
+      - run:
+          name: Create new release
+          command: |
+            if [ -n "$RELEASE_TYPE" ]; then
+              git config --global user.email "dev@purposeindustries.co"
+              git config --global user.name "purpose-bot"
+              git config --global push.default simple
+              npm version $RELEASE_TYPE --commit-hooks=false
+              git push -u origin $CIRCLE_BRANCH
+              git push --tags
             else
-              npm run test:e2e
+              echo "No RELEASE_TYPE is present, skipping creating new release."
+            fi
+      - run:
+          name: Publish new version to npm
+          command: |
+            # matches v1.2.3-next, or v1.2.3-1
+            if [[ "$CIRCLE_TAG" =~ ^v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(\-[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)+(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$ ]]; then
+              npm publish
+            else
+              echo "We are not building a tag or the tag is not a valid semver ($CIRCLE_TAG)";
             fi
+
+      - run:
+          name: Deploy the version
+          command: |
+            if [[ "$CIRCLE_TAG" =~ ^v[0-9]+(\.[0-9]+)*$ ]]; then
+              npm publish
+              npm run deploy
+            else
+              echo "We are not building a tag, or the tag is not major, minor or patch but preversion ($CIRCLE_TAG)"
+            fi
+
+  #Duplicated code. Not possible to solve with yaml references on circleci
+  nightly_build:
+    <<: *defaults
+    steps:
+      - checkout
+
+      - save_cache:
+          key: v1-repo-{{ .Environment.CIRCLE_SHA1 }}
```

Comment:
uh fuck this key is wrong. it should be `v1-repo-{{ checksum "package.json" }}`, can you replace everywhere?

Reply:
can you change this @molnosikristof?